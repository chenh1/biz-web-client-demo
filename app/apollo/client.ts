// import { ApolloLink, HttpLink } from "@apollo/client";
// import { onError } from "@apollo/client/link/error";

// import {
//   NextSSRInMemoryCache,
//   NextSSRApolloClient,
// } from "@apollo/experimental-nextjs-app-support/ssr";
// import { NextSSRInMemoryCache } from "@apollo/experimental-nextjs-app-support/ssr";
// import { registerApolloClient } from "@apollo/experimental-nextjs-app-support";}

// const errorLink = onError(({ graphQLErrors, networkError }) => {
//   if (graphQLErrors)
//     graphQLErrors.forEach(({ message, locations, path, ...rest }) => {
//       Object.keys(rest)?.map(key => console.log(key, rest[key]))
//       console.log(
//         `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
//       )
//     });
//   if (networkError) console.log(`[Network error]: ${networkError}`);
// });

// const httpLink = new HttpLink({
//   uri: `${process.env.API_URL || "http://127.0.0.1:1337"}/graphql`,
// })

// export const { getClient } = registerApolloClient(() => {
//   return new NextSSRApolloClient({
//     cache: new NextSSRInMemoryCache(),
//     link: ApolloLink.from([errorLink, httpLink]),
//   });
// });

import { HttpLink } from "@apollo/client";
import {
  registerApolloClient,
  ApolloClient,
  InMemoryCache,
} from "@apollo/experimental-nextjs-app-support";

export const { getClient, query, PreloadQuery } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      // this needs to be an absolute url, as relative urls cannot be used in SSR
      uri: `${process.env.API_URL || "http://127.0.0.1:1337"}/graphql`,
      // you can disable result caching here if you want to
      // (this does not work if you are rendering your page with `export const dynamic = "force-static"`)
      // fetchOptions: { cache: "no-store" },
    }),
  });
});