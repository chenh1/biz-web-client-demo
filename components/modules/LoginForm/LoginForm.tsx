"use client";

import type { UserFormSubmitValues } from "@/components/constants/types";
import React, { useState, useEffect } from "react"
import { useMutation } from "@apollo/client";
import { Box } from "../../containers/Box/Box";
import { Heading } from "../../core/Heading/Heading";
import { Button } from "../../core/Button/Button";
import { SpanLink } from "../../core/SpanLink/SpanLink";
import { InputText } from "../../fields/InputText/InputText";
import { Paragraph } from "../../core/Paragraph/Paragraph";
import { registerMutation } from "../../../graphql/mutations/register";
import { signIn } from "next-auth/react";
import { emailForgotPasswordMutation } from "../../../graphql/mutations/emailForgotPassword";
import { Checkbox } from "../../fields/Checkbox/Checkbox";

interface LoginFormProps {
  defaultShowResetPasswordPrompt?: boolean;
  defaultIsRegistering: boolean;
  darkModeBorder?: boolean;
  toggleVisibility: (boolean) => void;
}

export const LoginForm: React.FC<LoginFormProps> = ({ defaultShowResetPasswordPrompt=false, defaultIsRegistering, darkModeBorder, toggleVisibility }: LoginFormProps) => {
  //const client = getClient();
  
  const [ loginError, setLoginError ] = useState<string | null>(null)
  const [ forgotPasswordMessage, setForgotPasswordMessage ] = useState<string | null>(null)
  const [ forgotPasswordSuccessful, setForgotPasswordSuccessful ] = useState<boolean>(false)
  const [ formError, setFormError ] = useState<string | null>(null)
  const [ showPassword, setShowPassword ] = useState<boolean>(false);
  const [ nextAuthLoading, setNextAuthLoading ] = useState<boolean>(false);

  const [ register ] = useMutation(registerMutation);
  const [ forgotPassword, { data: forgotPasswordRes } ] = useMutation(emailForgotPasswordMutation, {
    onCompleted: ({ emailForgotPassword }) => {
      const { isSuccessful, status, statusText } = emailForgotPassword

        if (isSuccessful) {
          setForgotPasswordSuccessful(true)
        }

      if (!isSuccessful && status === 400) {
        const provider = statusText?.replace('Your login provider: ', '') || ''
        const providerName = `${provider.charAt(0).toUpperCase()}${provider.substr(1)}`
        setForgotPasswordMessage(`You signed in using ${providerName}. Close this window and try again.`)
      }
    }
  });


  
  const [ showResetPasswordPrompt, setShowResetPasswordPrompt ] = useState<boolean>(defaultShowResetPasswordPrompt);
  const [ doPasswordsMatch, setDoPasswordsMatch ] = useState<boolean>(true);

  const [ isRegistering, setIsRegistering ] = useState<boolean>(defaultIsRegistering)

  useEffect(() => {
    if (forgotPasswordSuccessful) {
      setTimeout(() => {
        setShowResetPasswordPrompt(false)
        setForgotPasswordSuccessful(false)
      }, 8000)
    }

    if (forgotPasswordMessage) {
      setTimeout(() => {
        setForgotPasswordMessage(null)
      }, 8000)
    }
  }, [forgotPasswordSuccessful, forgotPasswordMessage])

  const baseUrl = process.env.NEXT_PUBLIC_BASE_URL || 'https://biz-web-client-demo-static.s3.us-east-2.amazonaws.com'

  return (
    <Box padding={darkModeBorder ? "md" : "0"} className={`min-w-[300px] ${darkModeBorder ? "relative max-w-lg dark:border-2 dark:border-max-white" : ""}`} hasShadow={darkModeBorder} isRounded={darkModeBorder} gap="md">
      <Box padding="0">
        <Heading type="h4" className="text-center">Login</Heading>

        {/* login with google button*/}
        <div onClick={() => signIn('google')} className="justify-self-center cursor-pointer p-2 h-12 w-full sm:w-12 hover:w-full duration-300 ease-in-out overflow-hidden bg-apex-blue rounded-full">
          <Box padding="0" gap="sm" className="grid-cols-[auto_auto]">
            <Box padding="0" className="w-8 h-8">
              <img width={32} height={32} src={`${baseUrl}/google_icon_e7422bae08.png`} alt="Google icon"/>
            </Box>
            <Box padding="0" className="self-center">
              <Heading type="h4" fontSizeClass="text-xl" className="whitespace-nowrap">Login with Google</Heading>
            </Box>
          </Box>
        </div>
      </Box>
      

      {/* traditional login */}
      {showResetPasswordPrompt ?
        <form onSubmit={(e: UserFormSubmitValues) => {
          e.preventDefault()
          const email = e.target?.email?.value
          
          if (email) {
            forgotPassword({
              variables: {
                email: email,
              }
            })
          }
        }}>
          {forgotPasswordSuccessful ?
            <Box padding="0">
              <Paragraph>Reset link sent to email. Please reset your password and try logging in again.</Paragraph>
            </Box>
            : forgotPasswordMessage ?
              <Box padding="0">
                <Paragraph>You signed in with a Google account. Please click on the Google sign-in button right above this message!</Paragraph>
              </Box>
              :
              <Box padding="0" gap="md">
                <Paragraph size="sm" className="italic">Forgot your password? Enter the email you registered with the password you need to reset and we'll send you an email to update your password.</Paragraph>
                <InputText fullWidth id="email" name="email" placeholder="Email" />
                <Button type="submit" label={"Reset Password"} primary/>
                <a data-testid="cancelForgotPassword" className="text-center" href="javascript:void(0)" onClick={() => {
                  //setTimeout(() => {
                    setShowResetPasswordPrompt(false)
                  //}, 200)
                }}>
                  <SpanLink uncolored>
                    Nevermind, I remember my password
                  </SpanLink>
                </a>
              </Box>
          }
        </form>
        :
        <form onSubmit={(e: UserFormSubmitValues) => {
          e.preventDefault()
          
          const email = e.target?.email?.value
          const password = e.target?.password?.value
          const passwordConfirmation = e.target?.passwordConfirmation?.value
          
          if (isRegistering) {
            if (password !== passwordConfirmation) {
              setFormError("Passwords do not match")
            } else if (!email || !password || !passwordConfirmation) {
              setFormError("Please fill out all fields")
            } else {
              register({
                variables: {
                  username: email,
                  email: email,
                  password: password
                }
              }).then(({ data: { register: registerData } = {} }) => {
                if (registerData && registerData?.jwt) {
                  signIn(
                    'credentials', 
                    { redirect: false, email, password }
                  ).then((response) => {
                    if (response) {
                      const { status, ok, ...rest } = response;
                      if ((status === 401 && !ok) || rest.error === 'CredentialsSignin') {
                        setLoginError('Username and/or password is incorrect')
                      } else {
                        setFormError(null)
                        toggleVisibility(false)
                      }
                      setNextAuthLoading(false)
                    }
                  });
                }
              })
            }
          } else {
            signIn(
              'credentials', 
              { redirect: false, email, password }
            ).then((response) => {
              if (response) {
                const { status, ok, ...rest } = response;
                if ((status === 401 && !ok) || rest.error === 'CredentialsSignin') {
                  setFormError('Username and/or password is incorrect')
                } else {
                  setFormError(null)
                  toggleVisibility(false)
                  setTimeout(() => {
                    
                  }, 200);
                }
                setNextAuthLoading(false)
              }
            });
          }
        }}>
          <Box padding="0" gap={formError ? "micro" : "md"}>
            <Box padding="0" gap="xs">
              <Box padding="0" gap="md">
                <InputText dataCy="login-email-field" fullWidth id="email" name="email" placeholder="Email" />
                <InputText dataCy="login-password-field" fullWidth id="password" name="password" password={!showPassword} placeholder="Password" />
                {isRegistering && <InputText fullWidth id="password" name="passwordConfirmation" password={!showPassword} placeholder="Confirm Password" />}
              </Box>
              <Checkbox onChange={e => setShowPassword(e?.target?.checked)} label="Show password" /> 
            </Box>
            <Box padding="0" gap="md">
              <Box padding="0" gap="xs">
                {formError && <Paragraph textColor="night-black" darkTextColor="hyper-red" className="italic text-center" size="sm">{formError}</Paragraph>}
                <Button data-cy="login-register-cta" label={isRegistering ? "Register" : "Login"} type="submit" primary/>
                {!isRegistering && 
                  <a data-testid="openForgotPassword" className="text-center" href="javascript:void(0)" onClick={() => {
                    //setTimeout(() => {
                      setShowResetPasswordPrompt(true)
                    //}, 200)
                  }}>
                    <SpanLink uncolored>
                      Forgot your password? Click here
                    </SpanLink>
                  </a>
                }
              </Box>
              <Box padding="0">
                <Button onClick={() => setIsRegistering(!isRegistering)} label={isRegistering ? "Login instead" : "New here? Register now"}/>
              </Box>
            </Box>
          </Box>
        </form>
      }
    </Box>
  )
}