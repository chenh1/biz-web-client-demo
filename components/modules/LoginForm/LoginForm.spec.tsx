import { LoginForm } from "./LoginForm";
import { fireEvent, render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window

jest.mock("@apollo/client", () => ({
  useMutation: jest.fn(),
  gql: jest.fn()
}));

import { useMutation } from "@apollo/client";
const mockedUseMutation = useMutation as jest.Mock;

mockedUseMutation.mockImplementation((mutationDocument, onEvents) => {
  return [
    () => null,
    { data: {} }
  ]
})


it('renders LoginForm module', async () => {
  const { getByTestId, asFragment } = render(
    <LoginForm toggleVisibility={() => {}} defaultIsRegistering={false} />
  )

  expect(asFragment()).toMatchSnapshot();
});

it('shows registration form module', async () => {
  const { getByTestId, asFragment } = render(
    <LoginForm toggleVisibility={() => {}} defaultIsRegistering={true} />
  )

  expect(asFragment()).toMatchSnapshot();
});

it('shows reset form module and reverts back if cancelled', async () => {
  const { getByTestId, asFragment } = render(
    <LoginForm toggleVisibility={() => {}} defaultIsRegistering={false} />
  )

  fireEvent.click(getByTestId("openForgotPassword"))

  expect(asFragment()).toMatchSnapshot();

  fireEvent.click(getByTestId("cancelForgotPassword"))

  expect(asFragment()).toMatchSnapshot();
});
