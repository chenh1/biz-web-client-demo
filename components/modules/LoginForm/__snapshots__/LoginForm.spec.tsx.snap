// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginForm module 1`] = `
<DocumentFragment>
  <div
    class=" grid  gap-8  p-2 min-w-[300px] "
  >
    <div
      class=" grid  gap-4  p-2 "
    >
      <h4
        class="text-night-black dark:text-max-white font-headings font-semibold text-lg sm:text-md text-center"
      >
        Login
      </h4>
      <div
        class="justify-self-center cursor-pointer p-2 h-12 w-full sm:w-12 hover:w-full duration-300 ease-in-out overflow-hidden bg-apex-blue rounded-full"
      >
        <div
          class=" grid  gap-4  p-2 grid-cols-[auto_auto]"
        >
          <div
            class=" grid  gap-4  p-2 w-8 h-8"
          >
            <img
              alt="Google icon"
              height="32"
              src="https://biz-web-client-demo-static.s3.us-east-2.amazonaws.com/google_icon_e7422bae08.png"
              width="32"
            />
          </div>
          <div
            class=" grid  gap-4  p-2 self-center"
          >
            <h4
              class="text-night-black dark:text-max-white font-headings font-semibold text-xl whitespace-nowrap"
            >
              Login with Google
            </h4>
          </div>
        </div>
      </div>
    </div>
    <form>
      <div
        class=" grid  gap-8  p-2 "
      >
        <div
          class=" grid  gap-2  p-2 "
        >
          <div
            class=" grid  gap-8  p-2 "
          >
            <div
              class="relative "
            >
              <label
                class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
                for="email"
              >
                Email
              </label>
              <input
                class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
                data-cy="login-email-field"
                name="email"
                type="text"
              />
            </div>
            <div
              class="relative "
            >
              <label
                class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
                for="password"
              >
                Password
              </label>
              <input
                class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
                data-cy="login-password-field"
                name="password"
                type="password"
              />
            </div>
          </div>
          <span
            class="relative"
          >
            <input
              class="  h-0 w-0 absolute"
              data-testid="checkbox"
              id="tempId"
              type="checkbox"
              value=""
            />
            <label
              class="  font-body relative flex my-[.6em] items-center text-night-black dark:text-max-white transition-colors duration-[250ms] ease-[cubic-bezier(.4,.0,.23,1)]"
              for="tempId"
            >
              <span
                class="pointer-events-none flex justify-center items-center mr-[.9em] w-[1.1em] h-[1.1em] bg-transparent border-2 border-solid rounded-sm cursor-pointer transition-all duration-[250ms] ease-[cubic-bezier(.4,.0,.23,1)] border-night-black dark:border-max-white"
              />
              <span
                class="pointer-events-none duration-500 opacity-1"
              >
                Show password
              </span>
              <ins
                class="pointer-events-none absolute block bottom-0 left-[2em] h-0 w-full overflow-hidden no-underline transition-[height] duration-300 ease-[cubic-bezier(.4,.0,.23,1)] "
              >
                <i
                  class="pointer-events-none absolute bottom-0 not-italic text-hyper-red"
                >
                  Show password
                </i>
              </ins>
            </label>
          </span>
        </div>
        <div
          class=" grid  gap-8  p-2 "
        >
          <div
            class=" grid  gap-2  p-2 "
          >
            <button
              class="   w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-max-white px-5 py-2.5   bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red"
              data-cy="login-register-cta"
              data-testid="button"
              type="submit"
            >
              <div
                class="bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red rounded-lg absolute left-0 top-0 w-full h-full "
              />
              <span
                class="pointer-events-none relative z-10"
              >
                Login
              </span>
            </button>
            <a
              class="text-center"
              data-testid="openForgotPassword"
              href="javascript:void(0)"
            >
              <span
                class="  relative cursor-pointer font-body text-night-black dark:text-max-white before:content-[""] before:absolute before:translate-y-[6px] before:bottom-0 before:duration-300 before:ease-in-out before:border-apex-blue before:w-0 before:border-2 before:opacity-0 hover:brightness-110 hover:before:w-full hover:before:opacity-100"
              >
                Forgot your password? Click here
              </span>
            </a>
          </div>
          <div
            class=" grid  gap-4  p-2 "
          >
            <button
              class=" secondary  w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-slate-700 bg-transparent border-slate-700 dark:text-max-white dark:border-max-white active:text-max-white active:bg-slate-700 active:border-slate-700 dark:active:bg-max-white dark:active:border-max-white dark:active:text-night-black px-5 py-2.5   "
              data-testid="button"
              type="button"
            >
              <div
                class="triangle left"
              >
                <div />
              </div>
              <div
                class="triangle right"
              >
                <div />
              </div>
              <span
                class="pointer-events-none relative z-10"
              >
                New here? Register now
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`shows registration form module 1`] = `
<DocumentFragment>
  <div
    class=" grid  gap-8  p-2 min-w-[300px] "
  >
    <div
      class=" grid  gap-4  p-2 "
    >
      <h4
        class="text-night-black dark:text-max-white font-headings font-semibold text-lg sm:text-md text-center"
      >
        Login
      </h4>
      <div
        class="justify-self-center cursor-pointer p-2 h-12 w-full sm:w-12 hover:w-full duration-300 ease-in-out overflow-hidden bg-apex-blue rounded-full"
      >
        <div
          class=" grid  gap-4  p-2 grid-cols-[auto_auto]"
        >
          <div
            class=" grid  gap-4  p-2 w-8 h-8"
          >
            <img
              alt="Google icon"
              height="32"
              src="https://biz-web-client-demo-static.s3.us-east-2.amazonaws.com/google_icon_e7422bae08.png"
              width="32"
            />
          </div>
          <div
            class=" grid  gap-4  p-2 self-center"
          >
            <h4
              class="text-night-black dark:text-max-white font-headings font-semibold text-xl whitespace-nowrap"
            >
              Login with Google
            </h4>
          </div>
        </div>
      </div>
    </div>
    <form>
      <div
        class=" grid  gap-8  p-2 "
      >
        <div
          class=" grid  gap-2  p-2 "
        >
          <div
            class=" grid  gap-8  p-2 "
          >
            <div
              class="relative "
            >
              <label
                class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
                for="email"
              >
                Email
              </label>
              <input
                class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
                data-cy="login-email-field"
                name="email"
                type="text"
              />
            </div>
            <div
              class="relative "
            >
              <label
                class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
                for="password"
              >
                Password
              </label>
              <input
                class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
                data-cy="login-password-field"
                name="password"
                type="password"
              />
            </div>
            <div
              class="relative "
            >
              <label
                class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
                for="passwordConfirmation"
              >
                Confirm Password
              </label>
              <input
                class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
                name="passwordConfirmation"
                type="password"
              />
            </div>
          </div>
          <span
            class="relative"
          >
            <input
              class="  h-0 w-0 absolute"
              data-testid="checkbox"
              id="tempId"
              type="checkbox"
              value=""
            />
            <label
              class="  font-body relative flex my-[.6em] items-center text-night-black dark:text-max-white transition-colors duration-[250ms] ease-[cubic-bezier(.4,.0,.23,1)]"
              for="tempId"
            >
              <span
                class="pointer-events-none flex justify-center items-center mr-[.9em] w-[1.1em] h-[1.1em] bg-transparent border-2 border-solid rounded-sm cursor-pointer transition-all duration-[250ms] ease-[cubic-bezier(.4,.0,.23,1)] border-night-black dark:border-max-white"
              />
              <span
                class="pointer-events-none duration-500 opacity-1"
              >
                Show password
              </span>
              <ins
                class="pointer-events-none absolute block bottom-0 left-[2em] h-0 w-full overflow-hidden no-underline transition-[height] duration-300 ease-[cubic-bezier(.4,.0,.23,1)] "
              >
                <i
                  class="pointer-events-none absolute bottom-0 not-italic text-hyper-red"
                >
                  Show password
                </i>
              </ins>
            </label>
          </span>
        </div>
        <div
          class=" grid  gap-8  p-2 "
        >
          <div
            class=" grid  gap-2  p-2 "
          >
            <button
              class="   w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-max-white px-5 py-2.5   bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red"
              data-cy="login-register-cta"
              data-testid="button"
              type="submit"
            >
              <div
                class="bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red rounded-lg absolute left-0 top-0 w-full h-full "
              />
              <span
                class="pointer-events-none relative z-10"
              >
                Register
              </span>
            </button>
          </div>
          <div
            class=" grid  gap-4  p-2 "
          >
            <button
              class=" secondary  w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-slate-700 bg-transparent border-slate-700 dark:text-max-white dark:border-max-white active:text-max-white active:bg-slate-700 active:border-slate-700 dark:active:bg-max-white dark:active:border-max-white dark:active:text-night-black px-5 py-2.5   "
              data-testid="button"
              type="button"
            >
              <div
                class="triangle left"
              >
                <div />
              </div>
              <div
                class="triangle right"
              >
                <div />
              </div>
              <span
                class="pointer-events-none relative z-10"
              >
                Login instead
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`shows reset form module and reverts back if cancelled 1`] = `
<DocumentFragment>
  <div
    class=" grid  gap-8  p-2 min-w-[300px] "
  >
    <div
      class=" grid  gap-4  p-2 "
    >
      <h4
        class="text-night-black dark:text-max-white font-headings font-semibold text-lg sm:text-md text-center"
      >
        Login
      </h4>
      <div
        class="justify-self-center cursor-pointer p-2 h-12 w-full sm:w-12 hover:w-full duration-300 ease-in-out overflow-hidden bg-apex-blue rounded-full"
      >
        <div
          class=" grid  gap-4  p-2 grid-cols-[auto_auto]"
        >
          <div
            class=" grid  gap-4  p-2 w-8 h-8"
          >
            <img
              alt="Google icon"
              height="32"
              src="https://biz-web-client-demo-static.s3.us-east-2.amazonaws.com/google_icon_e7422bae08.png"
              width="32"
            />
          </div>
          <div
            class=" grid  gap-4  p-2 self-center"
          >
            <h4
              class="text-night-black dark:text-max-white font-headings font-semibold text-xl whitespace-nowrap"
            >
              Login with Google
            </h4>
          </div>
        </div>
      </div>
    </div>
    <form>
      <div
        class=" grid  gap-8  p-2 "
      >
        <p
          class=" text-night-black dark:text-max-white font-body font-body text-sm italic"
        >
          Forgot your password? Enter the email you registered with the password you need to reset and we'll send you an email to update your password.
        </p>
        <div
          class="relative "
        >
          <label
            class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
            for="email"
          >
            Email
          </label>
          <input
            class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
            name="email"
            type="text"
          />
        </div>
        <button
          class="   w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-max-white px-5 py-2.5   bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red"
          data-testid="button"
          type="submit"
        >
          <div
            class="bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red rounded-lg absolute left-0 top-0 w-full h-full "
          />
          <span
            class="pointer-events-none relative z-10"
          >
            Reset Password
          </span>
        </button>
        <a
          class="text-center"
          data-testid="cancelForgotPassword"
          href="javascript:void(0)"
        >
          <span
            class="  relative cursor-pointer font-body text-night-black dark:text-max-white before:content-[""] before:absolute before:translate-y-[6px] before:bottom-0 before:duration-300 before:ease-in-out before:border-apex-blue before:w-0 before:border-2 before:opacity-0 hover:brightness-110 hover:before:w-full hover:before:opacity-100"
          >
            Nevermind, I remember my password
          </span>
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`shows reset form module and reverts back if cancelled 2`] = `
<DocumentFragment>
  <div
    class=" grid  gap-8  p-2 min-w-[300px] "
  >
    <div
      class=" grid  gap-4  p-2 "
    >
      <h4
        class="text-night-black dark:text-max-white font-headings font-semibold text-lg sm:text-md text-center"
      >
        Login
      </h4>
      <div
        class="justify-self-center cursor-pointer p-2 h-12 w-full sm:w-12 hover:w-full duration-300 ease-in-out overflow-hidden bg-apex-blue rounded-full"
      >
        <div
          class=" grid  gap-4  p-2 grid-cols-[auto_auto]"
        >
          <div
            class=" grid  gap-4  p-2 w-8 h-8"
          >
            <img
              alt="Google icon"
              height="32"
              src="https://biz-web-client-demo-static.s3.us-east-2.amazonaws.com/google_icon_e7422bae08.png"
              width="32"
            />
          </div>
          <div
            class=" grid  gap-4  p-2 self-center"
          >
            <h4
              class="text-night-black dark:text-max-white font-headings font-semibold text-xl whitespace-nowrap"
            >
              Login with Google
            </h4>
          </div>
        </div>
      </div>
    </div>
    <form>
      <div
        class=" grid  gap-8  p-2 "
      >
        <div
          class=" grid  gap-2  p-2 "
        >
          <div
            class=" grid  gap-8  p-2 "
          >
            <div
              class="relative "
            >
              <label
                class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
                for="email"
              >
                Email
              </label>
              <input
                class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
                data-cy="login-email-field"
                name="email"
                type="text"
              />
            </div>
            <div
              class="relative "
            >
              <label
                class="  pointer-events-none font-body duration-300 absolute block text-md font-light text-night-black dark:text-max-white top-0 left-0  "
                for="password"
              >
                Password
              </label>
              <input
                class="  max-w-full w-full text-night-black dark:text-max-white font-body bg-transparent border-b-2 focus:outline-none border-night-black dark:border-max-white"
                data-cy="login-password-field"
                name="password"
                type="password"
              />
            </div>
          </div>
          <span
            class="relative"
          >
            <input
              class="  h-0 w-0 absolute"
              data-testid="checkbox"
              id="tempId"
              type="checkbox"
              value=""
            />
            <label
              class="  font-body relative flex my-[.6em] items-center text-night-black dark:text-max-white transition-colors duration-[250ms] ease-[cubic-bezier(.4,.0,.23,1)]"
              for="tempId"
            >
              <span
                class="pointer-events-none flex justify-center items-center mr-[.9em] w-[1.1em] h-[1.1em] bg-transparent border-2 border-solid rounded-sm cursor-pointer transition-all duration-[250ms] ease-[cubic-bezier(.4,.0,.23,1)] border-night-black dark:border-max-white"
              />
              <span
                class="pointer-events-none duration-500 opacity-1"
              >
                Show password
              </span>
              <ins
                class="pointer-events-none absolute block bottom-0 left-[2em] h-0 w-full overflow-hidden no-underline transition-[height] duration-300 ease-[cubic-bezier(.4,.0,.23,1)] "
              >
                <i
                  class="pointer-events-none absolute bottom-0 not-italic text-hyper-red"
                >
                  Show password
                </i>
              </ins>
            </label>
          </span>
        </div>
        <div
          class=" grid  gap-8  p-2 "
        >
          <div
            class=" grid  gap-2  p-2 "
          >
            <button
              class="   w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-max-white px-5 py-2.5   bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red"
              data-cy="login-register-cta"
              data-testid="button"
              type="submit"
            >
              <div
                class="bg-hyper-red border-hyper-red dark:bg-hyper-red dark:border-hyper-red rounded-lg absolute left-0 top-0 w-full h-full "
              />
              <span
                class="pointer-events-none relative z-10"
              >
                Login
              </span>
            </button>
            <a
              class="text-center"
              data-testid="openForgotPassword"
              href="javascript:void(0)"
            >
              <span
                class="  relative cursor-pointer font-body text-night-black dark:text-max-white before:content-[""] before:absolute before:translate-y-[6px] before:bottom-0 before:duration-300 before:ease-in-out before:border-apex-blue before:w-0 before:border-2 before:opacity-0 hover:brightness-110 hover:before:w-full hover:before:opacity-100"
              >
                Forgot your password? Click here
              </span>
            </a>
          </div>
          <div
            class=" grid  gap-4  p-2 "
          >
            <button
              class=" secondary  w-full sm:w-auto cursor-pointer border-2 font-bold leading-none inline-block rounded-lg text-center font-headings relative text-slate-700 bg-transparent border-slate-700 dark:text-max-white dark:border-max-white active:text-max-white active:bg-slate-700 active:border-slate-700 dark:active:bg-max-white dark:active:border-max-white dark:active:text-night-black px-5 py-2.5   "
              data-testid="button"
              type="button"
            >
              <div
                class="triangle left"
              >
                <div />
              </div>
              <div
                class="triangle right"
              >
                <div />
              </div>
              <span
                class="pointer-events-none relative z-10"
              >
                New here? Register now
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
